### PR TITLE
Fix PR agent workflow integration tests

### PR DESCRIPTION
**Pull request by mohavro**
## Summary
- align the PR Agent workflow name assertion with the actual workflow name
- clean up duplicated environment variable check to fix workflow test syntax
- keep workflow action validation allowing local actions and ensure tests run

## Testing
- pytest tests/integration/test_github_workflows.py -k "pr_agent_name or pr_agent_node_version" -q

[Codex Task](https://chatgpt.com/codex/tasks/task_e_6921c8cdd3d883249498a4f9fcbec228)

### **Auto-created Ticket**

[#257](https://github.com/mohavro/financial-asset-relationship-db/issues/257)

### **PR Type**
Bug fix, Tests

### **Description**
- Add assertion to validate PR Agent workflow name matches expected value

- Remove duplicated environment variable validation code block

- Fix formatting of print statement for maintainability check

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Test PR Agent Workflow"] -->|Add name validation| B["Assert workflow name equals PR Agent Workflow"]
  C["Duplicated env check code"] -->|Remove duplication| D["Single env validation block"]
  E["Malformed print statement"] -->|Fix formatting| F["Properly formatted output"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>test_github_workflows.py</strong><dd><code>Fix workflow tests and remove code duplication</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests/integration/test_github_workflows.py

<ul><li>Add assertion in <code>test_pr_agent_name()</code> to verify workflow name equals <br>"PR Agent Workflow"<br> <li> Remove duplicated environment variable validation code block that was <br>causing syntax errors<br> <li> Reformat print statement in <code>check_env_vars()</code> for proper code <br>formatting and readability</ul>

</details>

  </td>
  <td><a href="https://github.com/mohavro/financial-asset-relationship-db/pull/256/files#diff-beb22b64961fc1bd1db2d23ffc9993259440be38824ffd024d4836d3ec9ac1a3">+8/-9</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

## Summary by Sourcery

Tighten validation of the PR Agent GitHub workflow configuration and clean up maintainability checks in workflow integration tests.

Bug Fixes:
- Fix duplicated environment variable validation block that caused malformed maintainability output in workflow tests.

Enhancements:
- Improve maintainability warning output formatting for non-compliant workflow environment variables.

Tests:
- Add an explicit assertion that the PR Agent workflow name matches the expected configured value in integration tests.

---
## Summary by cubic
Fix PR Agent workflow integration tests by asserting the correct workflow name and consolidating env var casing checks into a single maintainability assertion. Add deeper config/workflow validation, full chunker removal checks, and core docs to prevent regressions.

- **Bug Fixes**
  - Assert workflow name equals "PR Agent Workflow".
  - Replace duplicated env validation with one assertion and clearer message for non-UPPER_CASE vars.
  - Add integration tests for PR-agent config and workflow integrity/security (no duplicate keys, third-party action pinning, no hardcoded secrets), docs and requirements, and verify removal of context_chunker, labeler.yml, and related references.

- **New Features**
  - Add core project docs (tech spec, coding guidelines, wiki v2) and a quick test execution guide with summaries.

<sup>Written for commit a0136159bdc6d4e68aa8af6113c50ab2c3c43b8d. Summary will update on new commits. <a href="https://cubic.dev/pr/DashFin-FarDb/financial-asset-relationship-db/pull/256">Review in cubic</a></sup>

___

## **CodeAnt-AI Description**
Add comprehensive integration tests for PR agent config, workflows, and chunker removal

### What Changed
- New tests validate .github/pr-agent-config.yml (structure, required fields, semantic versioning, sensible ranges, no hardcoded secrets) so configuration errors fail CI with clear messages
- New tests confirm context_chunker was fully removed (no file, no imports, no workflow references) and that related dependencies (tiktoken) and scripts/readme were cleaned up
- Workflow tests updated to detect duplicate YAML keys, enforce environment variable naming conventions with clearer maintainability messages, ensure secrets are referenced (not hardcoded), and verify simplified/safe workflow steps (no chunking steps, single Setup Python per job, pinned actions)
- requirements-dev.txt checks added/updated to require PyYAML presence and reasonable version constraints and to detect duplicate or conflicting dev dependency entries

### Impact
`✅ Fewer workflow regressions`
`✅ Clearer configuration validation failures in CI`
`✅ Fewer hardcoded secrets slipping into workflows`
<details>
<summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

File: tests/integration/test_github_workflows.py [hunk 1]
```diff
@@ -1190,15 +1193,11 @@ def check_env_vars(env_dict):
...
-                print(f"MAINTAINABILITY: Workflow {workflow_file.name} has environment variables "
-                      f"that don't follow UPPER_CASE convention: {invalid}. This can reduce "
-                      f"readability and consistency across workflows.")
-                      f"that don't follow UPPER_CASE convention: {invalid}. This can reduce "
-                      f"readability and consistency across workflows.")
+                print(
+                    f"MAINTAINABILITY: Workflow {workflow_file.name} has environment variables "
+                    f"that don't follow UPPER_CASE convention: {invalid}. This can reduce "
+                    f"readability and consistency across workflows."
+                )
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_github_workflows.py]: @codex fix this

**Review by mohavro — COMMENTED** [review]
Please address the comments from this code review:

## Overall Comments
- The new assertion that the workflow name equals the hard-coded string 'PR Agent Workflow' may be brittle if the workflow is ever renamed; consider sourcing the expected name from a shared constant or configuration so tests and workflows stay in sync.
- Instead of printing the MAINTAINABILITY warning in check_env_vars, consider surfacing it via pytest warnings or assertions so that non-conforming env vars are easier to track in automated test output and CI.

## Individual Comments

### Comment 1
<location> `tests/integration/test_github_workflows.py:1193-1200` </location>
<code_context>

</code_context>

<issue_to_address>
**issue (code-quality):** Avoid conditionals in tests. ([`no-conditionals-in-tests`](https://docs.sourcery.ai/Reference/Rules-and-In-Line-Suggestions/Python/Default-Rules/no-conditionals-in-tests))

<details><summary>Explanation</summary>Avoid complex code, like conditionals, in test functions.

Google's software engineering guidelines says:
"Clear tests are trivially correct upon inspection"
To reach that avoid complex code in tests:
* loops
* conditionals

Some ways to fix this:

* Use parametrized tests to get rid of the loop.
* Move the complex logic into helpers.
* Move the complex part into pytest fixtures.

> Complexity is most often introduced in the form of logic. Logic is defined via the imperative parts of programming languages such as operators, loops, and conditionals. When a piece of code contains logic, you need to do a bit of mental computation to determine its result instead of just reading it off of the screen. It doesn't take much logic to make a test more difficult to reason about.

Software Engineering at Google / [Don't Put Logic in Tests](https://abseil.io/resources/swe-book/html/ch12.html#donapostrophet_put_logic_in_tests)
</details>
</issue_to_address>

### Comment 2
<location> `tests/integration/test_github_workflows.py:1195-1200` </location>
<code_context>

</code_context>

<issue_to_address>
**issue (code-quality):** Avoid conditionals in tests. ([`no-conditionals-in-tests`](https://docs.sourcery.ai/Reference/Rules-and-In-Line-Suggestions/Python/Default-Rules/no-conditionals-in-tests))

<details><summary>Explanation</summary>Avoid complex code, like conditionals, in test functions.

Google's software engineering guidelines says:
"Clear tests are trivially correct upon inspection"
To reach that avoid complex code in tests:
* loops
* conditionals

Some ways to fix this:

* Use parametrized tests to get rid of the loop.
* Move the complex logic into helpers.
* Move the complex part into pytest fixtures.

> Complexity is most often introduced in the form of logic. Logic is defined via the imperative parts of programming languages such as operators, loops, and conditionals. When a piece of code contains logic, you need to do a bit of mental computation to determine its result instead of just reading it off of the screen. It doesn't take much logic to make a test more difficult to reason about.

Software Engineering at Google / [Don't Put Logic in Tests](https://abseil.io/resources/swe-book/html/ch12.html#donapostrophet_put_logic_in_tests)
</details>
</issue_to_address>

File: tests/integration/test_github_workflows.py [hunk 2]
```diff
@@ -1190,15 +1193,11 @@
...
-                print(f"MAINTAINABILITY: Workflow {workflow_file.name} has environment variables "
-                      f"that don't follow UPPER_CASE convention: {invalid}. This can reduce "
-                      f"readability and consistency across workflows.")
-                      f"that don't follow UPPER_CASE convention: {invalid}. This can reduce "
-                      f"readability and consistency across workflows.")
+                assert not invalid, (
+                    f"MAINTAINABILITY: Workflow {workflow_file.name} has environment variables "
+                    f"that don't follow UPPER_CASE convention: {invalid}. This can reduce "
+                    f"readability and consistency across workflows."
+                )
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_github_workflows.py]: @codex fix this

File: tests/integration/test_github_workflows.py [hunk 3]
```diff
@@ -306,6 +306,9 @@
         assert isinstance(pr_agent_workflow["name"], str) and pr_agent_workflow["name"].strip(), (
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_github_workflows.py]: @codex fix this

File: tests/integration/test_context_chunker_removal.py [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+            
+            try:
+                with open(py_file, 'r') as f:
+                    content = f.read()
+                
+                # Check for imports
+                assert 'import context_chunker' not in content, \
+                    f"{py_file} imports context_chunker"
+                assert 'from context_chunker' not in content, \
+                    f"{py_file} imports from context_chunker"
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_context_chunker_removal.py]: @codex fix this

File: tests/integration/test_documentation_validation.py
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_documentation_validation.py]: @qodo-merge-pro fix this

File: tests/integration/test_documentation_validation.py [hunk 5]
```diff
@@ -382,4 +382,268 @@ def test_consistent_line_endings(self):
...
+            if code_blocks:
+                # At least 80% should have language specified
+                with_language = sum(1 for lang in code_blocks if lang)
+                total = len(code_blocks)
+                
+                percentage = (with_language / total * 100) if total > 0 else 100
+                
+                assert percentage >= 70, \
+                    f"{doc_file}: Only {percentage:.0f}% of code blocks specify language. " \
+                    "Add language identifiers (bash, python, yaml, etc.)"
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_documentation_validation.py]: @qodo-merge-pro fix this

File: tests/integration/test_documentation_validation.py [hunk 6]
```diff
@@ -382,4 +382,271 @@ def test_consistent_line_endings(self):
...
+        content = summary_file.read_text()
+        
+        # Extract workflow file names mentioned
+        import re
+        workflow_mentions = re.findall(r'\.github/workflows/([a-z-]+\.yml)', content)
+        
+        for workflow_file in workflow_mentions:
+            workflow_path = Path(f'.github/workflows/{workflow_file}')
+            assert workflow_path.exists(), \
+                f"Documented workflow not found: {workflow_file}"
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_documentation_validation.py]: @qodo-merge-pro fix this

File: tests/integration/test_github_workflows.py
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_github_workflows.py]: @qodo-merge-pro fix this

File: tests/integration/test_requirements_dev.py [hunk 7]
```diff
@@ -333,4 +333,310 @@ def test_types_pyyaml_added(self, requirements: List[Tuple[str, str]]):
     def test_existing_packages_preserved(self, requirements: List[Tuple[str, str]]):
     def test_existing_packages_preserved(self, requirements: List[Tuple[str, str]]):
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_requirements_dev.py]: @qodo-merge-pro fix this

File: tests/integration/test_requirements_dev.py [hunk 8]
```diff
@@ -333,4 +333,310 @@ def test_types_pyyaml_added(self, requirements: List[Tuple[str, str]]):
...
         missing = [pkg for pkg in expected_packages if pkg not in package_names]
-        assert not missing, f"Missing expected packages: {missing}"
\ No newline at end of file
+        assert not missing, f"Missing expected packages: {missing}"
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_requirements_dev.py]: @qodo-merge-pro fix this

File: tests/integration/test_workflow_yaml_validation.py [hunk 9]
```diff
@@ -0,0 +1,493 @@
...
+        
+        for i, line in enumerate(lines, 1):
+            if line.strip() and not line.strip().startswith('#'):
+                # Count leading spaces
+                leading_spaces = len(line) - len(line.lstrip(' '))
+                
+                # Should be multiple of 2
+                if leading_spaces > 0:
+                    assert leading_spaces % 2 == 0, \
+                        f"{workflow_file.name}:{i} has inconsistent indentation ({leading_spaces} spaces)"
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_workflow_yaml_validation.py]: @qodo-merge-pro fix this

File: tests/integration/test_workflow_yaml_validation.py [hunk 10]
```diff
@@ -0,0 +1,494 @@
...
+                    
+                    # Extract version
+                    version = action.split('@')[1]
+                    
+                    # Warn if using moving tags like @main or @master
+                    moving_tags = ['main', 'master', 'latest', 'stable']
+                    if version in moving_tags:
+                        # This is a warning, not an error - but we'll document it
+                        pass  # Could add logging here
+
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_workflow_yaml_validation.py]: @qodo-merge-pro fix this

File: tests/integration/test_context_chunker_removal.py [hunk 11]
```diff
@@ -0,0 +1,338 @@
...
+        
+        for workflow_file in workflow_files:
+            with open(workflow_file, 'r') as f:
+                content = f.read()
+            
+            assert 'context_chunker.py' not in content, \
+                f"{workflow_file.name} references context_chunker.py"
+            assert 'context_chunker' not in content.lower() and 'context chunker' not in content.lower(), \
+                f"{workflow_file.name} may reference context chunking functionality"
+
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_context_chunker_removal.py]: @qodo-merge-pro fix this

File: tests/integration/test_context_chunker_removal.py [hunk 12]
```diff
@@ -0,0 +1,342 @@
...
+                content = f.read()
+            
+            assert 'context_chunker.py' not in content, \
+                f"{workflow_file.name} references context_chunker.py"
+            assert 'context_chunker' not in content.lower() and 'context chunker' not in content.lower(), \
+                f"{workflow_file.name} may reference context chunking functionality"
+import pytest
+from pathlib import Path
+import re
+import yaml
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_context_chunker_removal.py]: @qodo-merge-pro review this comment and resolve if it requires a fix.

**Review by mohavro — COMMENTED** [review]
final reviews before merge commit

**Review by mohavro — COMMENTED** [review]
Review all outstanding comments, make changes and fix all issues. PR will remain blocked from merging until all issues resolved.

File: tests/integration/test_github_workflows.py [hunk 13]
```diff
@@ -2557,207 +2643,344 @@ def test_pr_agent_setup_python_has_required_parameters(self):
...
-                    # Version should not be 'latest' or 'main' or 'master'
-                    assert version.lower() not in ['latest', 'main', 'master'], (
-                        f"Step #{idx} in job '{job_name}' uses floating version '{version}' for {action}. "
-                        "Use specific version tags (e.g., @v4) for reproducibility."
-                    )
+        # Should contain testing/linting tools
+        dev_tools = ['pytest', 'flake8', 'black', 'pyyaml', 'bandit', 'safety']
+        
+        found_tools = [tool for tool in dev_tools if tool in content]
+        assert len(found_tools) >= 2, "Should have at least 2 dev tools"
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_github_workflows.py]: @codeant-ai review the following and make changes as required:

In @tests/integration/test_github_workflows.py around lines 2738 - 2854,
TestWorkflowSecurityHardening is structurally corrupted: interleaved/fractured
test code must be cleaned and the misplaced PR-agent setup checks isolated into
their own test. Remove the orphaned/misplaced blocks inside
TestWorkflowSecurityHardening and restore each test to a single coherent body:
keep test_no_conflicting_dependencies focused on requirements-dev.txt parsing
(use req_file, content, lines, packages logic only) and move the setup-python
validation loop (references: test_pr_agent_setup_python_has_required_parameters,
pr_agent_file, load_yaml_safe, jobs, steps, uses, step.get("with"),
python-version parsing/regex) into a separate test function named
test_pr_agent_setup_python_has_required_parameters; ensure the new test reads
pr_agent_file via load_yaml_safe and validates python-version presence, format
and minimum version >= 3.8, and remove any duplicated req_file existence checks
from the mislocated block so each test has its own appropriate file checks.

File: tests/integration/test_github_workflows.py [hunk 14]
```diff
@@ -2557,207 +2643,344 @@ def test_pr_agent_setup_python_has_required_parameters(self):
...
-            if setup_python_by_name > 0 or setup_python_by_uses > 0:
-                assert setup_python_by_name == setup_python_by_uses, (
-                    f"Job '{job_name}' has mismatched Setup Python steps: "
-                    f"{setup_python_by_name} by name vs {setup_python_by_uses} by uses"
-                )
+            # Should have reasonable structure
+            lines = content.split('\n')
+            assert len(lines) > 10, f"{doc_file} is too short"
+
+
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_github_workflows.py]: @codeant-ai review the following comment and make changes to resolve issue: 

In @tests/integration/test_github_workflows.py around lines 2856 - 2915, The
TestWorkflowDocumentationConsistency class has its
test_documentation_files_valid_markdown method corrupted: the doc_files list is
interrupted by unrelated YAML validation code
(load_yaml_safe/pr_agent_file/workflow/jobs loop) which breaks structure;
restore doc_files to be a simple list of markdown filenames and move the YAML
validation block into its own test (e.g.,
test_pr_agent_workflow_setup_python_once) that uses
load_yaml_safe(pr_agent_file) to load the workflow, iterates
workflow.get("jobs", {}) and for each job checks steps to compute name_count and
uses_count and asserts the <=1 and equality conditions; ensure
test_documentation_files_valid_markdown only reads and validates markdown files
(non-empty, contains '#', length > 10) and that the new YAML test is correctly
indented inside the class as a separate method referencing pr_agent_file,
load_yaml_safe, workflow, jobs, job_name, steps.

File: tests/integration/test_github_workflows.py [hunk 15]
```diff
@@ -2557,207 +2643,344 @@ def test_pr_agent_setup_python_has_required_parameters(self):
...
+        # Should contain testing/linting tools
+        dev_tools = ['pytest', 'flake8', 'black', 'pyyaml', 'bandit', 'safety']
+        
+        found_tools = [tool for tool in dev_tools if tool in content]
+        assert len(found_tools) >= 2, "Should have at least 2 dev tools"
 
 
-class TestWorkflowChangeRegression:
-    """Regression tests specifically for the duplicate Setup Python fix."""
+class TestWorkflowDocumentationConsistency:
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_github_workflows.py]: @qodo-merge-pro fix this

File: tests/integration/test_github_workflows.py [hunk 16]
```diff
@@ -2517,33 +2544,92 @@ def parse_requirements(file_path):
...
+        
+        assert 'on' in pr_agent_workflow or True in pr_agent_workflow, "No triggers defined"
+        triggers = pr_agent_workflow.get('on', pr_agent_workflow.get(True, {}))
+        
+        # Should trigger on pull request review events
+        assert 'pull_request_review' in triggers or \
+               'pull_request' in triggers or \
+               'issue_comment' in triggers, \
+            "No appropriate PR-related triggers found"
+
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_github_workflows.py]: @codeant-ai review comment and make changes if needed.
